fineract-provider/src/main/java/org/apache/fineract/useradministration/starter/UserAdministrationAutoConfiguration.java

```diff
@@ -0,0 +1,101 @@
+package org.apache.fineract.useradministration.starter;
+
+import org.apache.fineract.infrastructure.core.service.database.DatabaseSpecificSQLGenerator;
+import org.apache.fineract.infrastructure.security.service.PlatformPasswordEncoder;
+import org.apache.fineract.infrastructure.security.service.PlatformSecurityContext;
+import org.apache.fineract.organisation.office.domain.OfficeRepositoryWrapper;
+import org.apache.fineract.organisation.office.service.OfficeReadPlatformService;
+import org.apache.fineract.organisation.staff.domain.StaffRepositoryWrapper;
+import org.apache.fineract.organisation.staff.service.StaffReadPlatformService;
+import org.apache.fineract.portfolio.client.domain.ClientRepositoryWrapper;
+import org.apache.fineract.useradministration.data.PasswordPreferencesDataValidator;
+import org.apache.fineract.useradministration.domain.AppUserPreviousPasswordRepository;
+import org.apache.fineract.useradministration.domain.AppUserRepository;
+import org.apache.fineract.useradministration.domain.PasswordValidationPolicyRepository;
+import org.apache.fineract.useradministration.domain.PermissionRepository;
+import org.apache.fineract.useradministration.domain.RoleRepository;
+import org.apache.fineract.useradministration.domain.UserDomainService;
+import org.apache.fineract.useradministration.serialization.PermissionsCommandFromApiJsonDeserializer;
+import org.apache.fineract.useradministration.service.AppUserReadPlatformServiceImpl;
+import org.apache.fineract.useradministration.service.AppUserWritePlatformServiceJpaRepositoryImpl;
+import org.apache.fineract.useradministration.service.PasswordPreferencesWritePlatformServiceJpaRepositoryImpl;
+import org.apache.fineract.useradministration.service.PasswordValidationPolicyReadPlatformServiceImpl;
+import org.apache.fineract.useradministration.service.PermissionReadPlatformServiceImpl;
+import org.apache.fineract.useradministration.service.PermissionWritePlatformServiceJpaRepositoryImpl;
+import org.apache.fineract.useradministration.service.RoleDataValidator;
+import org.apache.fineract.useradministration.service.RoleReadPlatformService;
+import org.apache.fineract.useradministration.service.RoleReadPlatformServiceImpl;
+import org.apache.fineract.useradministration.service.RoleWritePlatformServiceJpaRepositoryImpl;
+import org.apache.fineract.useradministration.service.UserDataValidator;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.jdbc.core.JdbcTemplate;
+
+@Configuration
+public class UserAdministrationAutoConfiguration {
+
+    @Bean
+    @ConditionalOnMissingBean
+    public AppUserReadPlatformServiceImpl appUserReadPlatformService(PlatformSecurityContext context, JdbcTemplate jdbcTemplate,
```

Comment:
Don't use implementation classes here, always use Java interfaces... they are not excheangable! Please provide the Java interface class reference in parenthesis with the `@ConditionalOnMissingBean` annotation.

Reply:
Done.